app/src/main/java/org/schabi/newpipe/info_list/StreamSegmentItem.kt

```diff
@@ -0,0 +1,59 @@
+package org.schabi.newpipe.info_list
+
+import android.widget.ImageView
+import android.widget.TextView
+import com.nostra13.universalimageloader.core.ImageLoader
+import com.xwray.groupie.GroupieViewHolder
+import com.xwray.groupie.Item
+import org.schabi.newpipe.R
+import org.schabi.newpipe.extractor.stream.StreamSegment
+import org.schabi.newpipe.util.ImageDisplayConstants
+import java.util.concurrent.TimeUnit
+
+class StreamSegmentItem(
+    private val item: StreamSegment,
+    private val onClick: StreamSegmentAdapter.StreamSegmentListener
+) : Item<GroupieViewHolder>() {
+
+    companion object {
+        const val PAYLOAD_SELECT = 1
+    }
+
+    var isSelected = false
+
+    override fun bind(viewHolder: GroupieViewHolder, position: Int) {
+        item.previewUrl?.let {
+            ImageLoader.getInstance().displayImage(
+                it, viewHolder.root.findViewById<ImageView>(R.id.previewImage),
+                ImageDisplayConstants.DISPLAY_THUMBNAIL_OPTIONS
+            )
+        }
+        viewHolder.root.findViewById<TextView>(R.id.textViewTitle).text = item.title
+        viewHolder.root.findViewById<TextView>(R.id.textViewStartSeconds).text =
+            secondsToString(item.startTimeSeconds.toLong())
+        viewHolder.root.setOnClickListener { onClick.onItemClick(this, item.startTimeSeconds) }
+        viewHolder.root.isSelected = isSelected
+    }
+
+    override fun bind(viewHolder: GroupieViewHolder, position: Int, payloads: MutableList<Any>) {
+        if (payloads.contains(PAYLOAD_SELECT)) {
+            viewHolder.root.isSelected = isSelected
+            return
+        }
+        super.bind(viewHolder, position, payloads)
+    }
+
+    private fun secondsToString(seconds: Long): String {
+        val hours = TimeUnit.SECONDS.toHours(seconds)
+        val minutes = TimeUnit.SECONDS.toMinutes(seconds)
+            .minus(TimeUnit.HOURS.toMinutes(hours))
+        val sec = seconds
+            .minus(TimeUnit.HOURS.toSeconds(hours))
+            .minus(TimeUnit.MINUTES.toSeconds(minutes))
+
+        return if (hours == 0L) String.format("%02d:%02d", minutes, sec)
+        else String.format("%d:%02d:%02d", hours, minutes, sec)
+    }
```

Comment:
This can probably be moved somewhere else, like in `LocalizationUtils` (or whatever that file is called). And a localized & standard timestamp should probably be generated by building a `LocalDateTime` and then formatting it using one of the default option. I'm not completely sure about this though, @B0pol do you have any opinion?

Reply:
I've looked it up and there is a method in the utils package which does the job (`Localization#getDurationString(final long duration)`). I'll use that method instead.